### ISSUE TITLE
Cannot call class as a function in v1.0.6 with react-slick <Slider> Component

### ISSUE THREAD

**Issue by khaledosman** [maicki/why-did-you-update#48]
Works with the previous why-did-you-update version
current versions:
"why-did-you-update": "1.0.6"
"react": "^16.6.3",
"react-dom": "^16.6.3",
 "react-slick": "^0.23.2",
```

inner-slider.js:78 Uncaught TypeError: Cannot call a class as a function
    at _classCallCheck (inner-slider.js:78)
    at InnerSlider (inner-slider.js:110)
    at WDYUFunctionalComponent.render (index.js:125)
    at finishClassComponent (react-dom.development.js:14899)
    at updateClassComponent (react-dom.development.js:14863)
    at beginWork (react-dom.development.js:15716)
    at performUnitOfWork (react-dom.development.js:18750)
    at workLoop (react-dom.development.js:18791)
    at HTMLUnknownElement.callCallback (react-dom.development.js:147)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:196)
    at invokeGuardedCallback (react-dom.development.js:250)
    at replayUnitOfWork (react-dom.development.js:17998)
    at renderRoot (react-dom.development.js:18909)
    at performWorkOnRoot (react-dom.development.js:19812)
    at performWork (react-dom.development.js:19722)
    at performSyncWork (react-dom.development.js:19696)
    at requestWork (react-dom.development.js:19551)
    at scheduleWork (react-dom.development.js:19358)
    at Object.enqueueSetState (react-dom.development.js:12789)
    at WDYUClassComponent.push../node_modules/react/cjs/react.development.js.Component.setState (react.development.js:354)
    at WDYUClassComponent.onStateChange (connectAdvanced.js:205)
    at Object.dispatch (redux.js:214)
    at dispatch (<anonymous>:1:38308)
    at redux-logger.js:458
    at index.js:11
    at dispatch (redux.js:613)
    at _callee2$ (index.js:81)
    at tryCatch (runtime.js:62)
    at Generator.invoke [as _invoke] (runtime.js:288)
    at Generator.prototype.(:3000/anonymous function) [as next] (http://localhost:3000/static/js/0.chunk.js:690:21)
    at asyncGeneratorStep (asyncToGenerator.js:3)
    at _next (asyncToGenerator.js:25)
_classCallCheck @ inner-slider.js:78
InnerSlider @ inner-slider.js:110
render @ index.js:125
finishClassComponent @ react-dom.development.js:14899
updateClassComponent @ react-dom.development.js:14863
beginWork @ react-dom.development.js:15716
performUnitOfWork @ react-dom.development.js:18750
workLoop @ react-dom.development.js:18791
callCallback @ react-dom.development.js:147
invokeGuardedCallbackDev @ react-dom.development.js:196
invokeGuardedCallback @ react-dom.development.js:250
replayUnitOfWork @ react-dom.development.js:17998
renderRoot @ react-dom.development.js:18909
performWorkOnRoot @ react-dom.development.js:19812
performWork @ react-dom.development.js:19722
performSyncWork @ react-dom.development.js:19696
requestWork @ react-dom.development.js:19551
scheduleWork @ react-dom.development.js:19358
enqueueSetState @ react-dom.development.js:12789
push../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:354
onStateChange @ connectAdvanced.js:205
dispatch @ redux.js:214
dispatch @ VM240:1
(anonymous) @ redux-logger.js:458
(anonymous) @ index.js:11
dispatch @ redux.js:613
_callee2$ @ index.js:81
tryCatch @ runtime.js:62
invoke @ runtime.js:288
prototype.(anonymous function) @ runtime.js:114
asyncGeneratorStep @ asyncToGenerator.js:3
_next @ asyncToGenerator.js:25
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js:13
_next @ asyncToGenerator.js:25
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js:13
_next @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
(anonymous) @ index.js:66
(anonymous) @ index.js:8
dispatch @ VM240:1
(anonymous) @ redux.js:463
componentDidMount @ App.js:74
commitLifeCycles @ react-dom.development.js:16722
commitAllLifeCycles @ react-dom.development.js:18160
callCallback @ react-dom.development.js:147
invokeGuardedCallbackDev @ react-dom.development.js:196
invokeGuardedCallback @ react-dom.development.js:250
commitRoot @ react-dom.development.js:18365
completeRoot @ react-dom.development.js:19894
performWorkOnRoot @ react-dom.development.js:19817
performWork @ react-dom.development.js:19722
performSyncWork @ react-dom.development.js:19696
requestWork @ react-dom.development.js:19551
scheduleWork @ react-dom.development.js:19358
enqueueSetState @ react-dom.development.js:12789
push../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:354
PersistGate._this.handlePersistorState @ react.js:86
dispatch @ redux.js:214
rehydrate @ persistStore.js:90
_rehydrate @ persistReducer.js:76
(anonymous) @ persistReducer.js:98
Promise.then (async)
(anonymous) @ persistReducer.js:97
Promise.then (async)
(anonymous) @ persistReducer.js:92
computeNextEntry @ VM240:1
recomputeStates @ VM240:1
(anonymous) @ VM240:1
dispatch @ redux.js:205
dispatch @ VM240:1
(anonymous) @ redux-logger.js:458
(anonymous) @ index.js:11
dispatch @ VM240:1
persist @ persistStore.js:125
persistStore @ persistStore.js:133
./src/configureStore.js @ configureStore.js:18
__webpack_require__ @ bootstrap:782
fn @ bootstrap:150
./src/actions/index.js @ QlairUtil.js:12
__webpack_require__ @ bootstrap:782
fn @ bootstrap:150
./src/App.js @ App.css?da7c:45
__webpack_require__ @ bootstrap:782
fn @ bootstrap:150
./src/index.js @ index.css?02e3:45
__webpack_require__ @ bootstrap:782
fn @ bootstrap:150
0 @ nativeAlert.js:12
__webpack_require__ @ bootstrap:782
checkDeferredModules @ bootstrap:45
webpackJsonpCallback @ bootstrap:32
(anonymous) @ main.chunk.js:1
index.js:1452 The above error occurred in the <InnerSlider> component:
    in InnerSlider (created by Slider)
    in Slider (at BuildingScreen.js:219)
    in div (at BuildingScreen.js:217)
    in BuildingScreen (created by Connect(BuildingScreen))
    in Connect(BuildingScreen) (created by Route)
    in Route (created by withRouter(Connect(BuildingScreen)))
    in withRouter(Connect(BuildingScreen)) (at ProtectedRoute.js:18)
    in Route (at ProtectedRoute.js:29)
    in ProtectedRoute (at App.js:42)
    in Switch (at App.js:39)
    in main (at App.js:38)
    in div (at App.js:36)
    in App (created by Connect(App))
    in Connect(App) (created by Route)
    in Route (created by withRouter(Connect(App)))
    in withRouter(Connect(App)) (at src/index.js:47)
    in Router (created by HashRouter)
    in HashRouter (at src/index.js:46)
    in PersistGate (at src/index.js:45)
    in Provider (at src/index.js:44)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
```

**Comment by borisyordanov** [comment]
I have the same bug and am using the same packages

**Comment by borisyordanov** [comment]
@danburzo i'm using typescript and checked my tsconfig, the target was es5, when i changed it to es6 i started getting these errors:
```
Uncaught TypeError: Class constructor App cannot be invoked without 'new'
    at new WDYUClassComponent (index.js:63)
    at constructClassInstance (react-dom.development.js:12484)
    at updateClassComponent (react-dom.development.js:14255)
    at beginWork (react-dom.development.js:15082)
    at performUnitOfWork (react-dom.development.js:17820)
    at workLoop (react-dom.development.js:17860)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)
    at replayUnitOfWork (react-dom.development.js:17107)
    at renderRoot (react-dom.development.js:17979)
    at performWorkOnRoot (react-dom.development.js:18837)
    at performWork (react-dom.development.js:18749)
    at performSyncWork (react-dom.development.js:18723)
    at requestWork (react-dom.development.js:18592)
    at scheduleWork (react-dom.development.js:18401)
    at scheduleRootUpdate (react-dom.development.js:19069)
    at updateContainerAtExpirationTime (react-dom.development.js:19097)
    at updateContainer (react-dom.development.js:19154)
    at ReactRoot../node_modules/react-dom/cjs/react-dom.development.js.ReactRoot.render (react-dom.development.js:19416)
    at react-dom.development.js:19556
    at unbatchedUpdates (react-dom.development.js:18952)
    at legacyRenderSubtreeIntoContainer (react-dom.development.js:19552)
    at Object.render (react-dom.development.js:19613)
    at Object../src/index.tsx (index.tsx:16)
    at __webpack_require__ (bootstrap 0a1c8fbda5af407dc916:678)
    at fn (bootstrap 0a1c8fbda5af407dc916:88)
    at Object.0 (variables.css?43e7:26)
    at __webpack_require__ (bootstrap 0a1c8fbda5af407dc916:678)
    at bootstrap 0a1c8fbda5af407dc916:724
    at bootstrap 0a1c8fbda5af407dc916:724
App @ index.js:63
constructClassInstance @ react-dom.development.js:12484
updateClassComponent @ react-dom.development.js:14255
beginWork @ react-dom.development.js:15082
performUnitOfWork @ react-dom.development.js:17820
workLoop @ react-dom.development.js:17860
callCallback @ react-dom.development.js:149
invokeGuardedCallbackDev @ react-dom.development.js:199
invokeGuardedCallback @ react-dom.development.js:256
replayUnitOfWork @ react-dom.development.js:17107
renderRoot @ react-dom.development.js:17979
performWorkOnRoot @ react-dom.development.js:18837
performWork @ react-dom.development.js:18749
performSyncWork @ react-dom.development.js:18723
requestWork @ react-dom.development.js:18592
scheduleWork @ react-dom.development.js:18401
scheduleRootUpdate @ react-dom.development.js:19069
updateContainerAtExpirationTime @ react-dom.development.js:19097
updateContainer @ react-dom.development.js:19154
./node_modules/react-dom/cjs/react-dom.development.js.ReactRoot.render @ react-dom.development.js:19416
(anonymous) @ react-dom.development.js:19556
unbatchedUpdates @ react-dom.development.js:18952
legacyRenderSubtreeIntoContainer @ react-dom.development.js:19552
render @ react-dom.development.js:19613
./src/index.tsx @ index.tsx:16
__webpack_require__ @ bootstrap 0a1c8fbda5af407dc916:678
fn @ bootstrap 0a1c8fbda5af407dc916:88
0 @ variables.css?43e7:26
__webpack_require__ @ bootstrap 0a1c8fbda5af407dc916:678
(anonymous) @ bootstrap 0a1c8fbda5af407dc916:724
(anonymous) @ bootstrap 0a1c8fbda5af407dc916:724
index.js:2178 


The above error occurred in the <App> component:
    in App

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
__stack_frame_overlay_proxy_console__ @ index.js:2178
logCapturedError @ react-dom.development.js:15749
logError @ react-dom.development.js:15783
update.callback @ react-dom.development.js:16649
callCallback @ react-dom.development.js:11164
commitUpdateEffects @ react-dom.development.js:11203
commitUpdateQueue @ react-dom.development.js:11194
commitLifeCycles @ react-dom.development.js:16010
commitAllLifeCycles @ react-dom.development.js:17262
callCallback @ react-dom.development.js:149
invokeGuardedCallbackDev @ react-dom.development.js:199
invokeGuardedCallback @ react-dom.development.js:256
commitRoot @ react-dom.development.js:17458
completeRoot @ react-dom.development.js:18912
performWorkOnRoot @ react-dom.development.js:18841
performWork @ react-dom.development.js:18749
performSyncWork @ react-dom.development.js:18723
requestWork @ react-dom.development.js:18592
scheduleWork @ react-dom.development.js:18401
scheduleRootUpdate @ react-dom.development.js:19069
updateContainerAtExpirationTime @ react-dom.development.js:19097
updateContainer @ react-dom.development.js:19154
./node_modules/react-dom/cjs/react-dom.development.js.ReactRoot.render @ react-dom.development.js:19416
(anonymous) @ react-dom.development.js:19556
unbatchedUpdates @ react-dom.development.js:18952
legacyRenderSubtreeIntoContainer @ react-dom.development.js:19552
render @ react-dom.development.js:19613
./src/index.tsx @ index.tsx:16
__webpack_require__ @ bootstrap 0a1c8fbda5af407dc916:678
fn @ bootstrap 0a1c8fbda5af407dc916:88
0 @ variables.css?43e7:26
__webpack_require__ @ bootstrap 0a1c8fbda5af407dc916:678
(anonymous) @ bootstrap 0a1c8fbda5af407dc916:724
(anonymous) @ bootstrap 0a1c8fbda5af407dc916:724
index.js:63 


Uncaught TypeError: Class constructor App cannot be invoked without 'new'
    at new WDYUClassComponent (index.js:63)
    at constructClassInstance (react-dom.development.js:12484)
    at updateClassComponent (react-dom.development.js:14255)
    at beginWork (react-dom.development.js:15082)
    at performUnitOfWork (react-dom.development.js:17820)
    at workLoop (react-dom.development.js:17860)
    at renderRoot (react-dom.development.js:17946)
    at performWorkOnRoot (react-dom.development.js:18837)
    at performWork (react-dom.development.js:18749)
    at performSyncWork (react-dom.development.js:18723)
    at requestWork (react-dom.development.js:18592)
    at scheduleWork (react-dom.development.js:18401)
    at scheduleRootUpdate (react-dom.development.js:19069)
    at updateContainerAtExpirationTime (react-dom.development.js:19097)
    at updateContainer (react-dom.development.js:19154)
    at ReactRoot../node_modules/react-dom/cjs/react-dom.development.js.ReactRoot.render (react-dom.development.js:19416)
    at react-dom.development.js:19556
    at unbatchedUpdates (react-dom.development.js:18952)
    at legacyRenderSubtreeIntoContainer (react-dom.development.js:19552)
    at Object.render (react-dom.development.js:19613)
    at Object../src/index.tsx (index.tsx:16)
    at __webpack_require__ (bootstrap 0a1c8fbda5af407dc916:678)
    at fn (bootstrap 0a1c8fbda5af407dc916:88)
    at Object.0 (variables.css?43e7:26)
    at __webpack_require__ (bootstrap 0a1c8fbda5af407dc916:678)
    at bootstrap 0a1c8fbda5af407dc916:724
    at bootstrap 0a1c8fbda5af407dc916:724
```

If i disable the `whyDidYouUpdate(React);` the app starts working as normal.

**Comment by khaledosman** [comment]
@danburzo I'm using create-react-app  / react-scripts v2, not sure exactly what's the target configured in webpack/babel, but I'm guessing it could be es5 for wider browser support

**Comment by vzaidman** [comment]
the reason is that ReactSlider uses render like this:
`render = () => {` 
and not
`render(){`
and it makes us think it's a functional component.

**Comment by vzaidman** [comment]
fixed this on my more advanced project
```
"@welldone-software/why-did-you-render": "2.2.3",
```
i would welcome a PR on this project that fixes the same issue.

you can look here to understand how:
https://github.com/welldone-software/why-did-you-render/blob/master/src/whyDidYouRender.js

**Comment by vzaidman** [comment]
more about this issue:
https://overreacted.io/how-does-react-tell-a-class-from-a-function/

**Comment by matthew-dean** [comment]
I'm getting the same issue, but not with `react-slider`, just in our application code. This module worked in the past.

**Comment by matthew-dean** [comment]
For me, `why-did-you-render` stops the crash but ends up never logging anything.

**Comment by matthew-dean** [comment]
Oh, I see, an explicit include like `{include: [/.*/]}` is now required?

Now I get `TypeError: Class constructor ReactiveHOC cannot be invoked without 'new'` in the minified lib. Related to: https://github.com/maicki/why-did-you-update/issues/47 ?

**Comment by AntonioRedondo** [comment]
Any update on this? On our project we extensively use `render = () => { ... }`. As long as we change to `render() { ... }` `why-did-you-update` works again. But we cannot make such a change as there are dozens of compoents.

**Comment by vzaidman** [comment]
You can mimic what i did in https://github.com/welldone-software/why-did-you-render/blob/master/src/whyDidYouRender.js as i mentioned or try using it.

**Comment by vzaidman** [comment]
I found out the problem is caused by transpiled class trying to extend native class.
It happens when you build your app with the target of latest browsers.
Targeting latest browsers cause babel-env to not transpile classes.
The library always transpiles classes.
Transpiled classes are failing to extend untranspiled classes.

Fixed this in the `why-did-you-render` library:
https://github.com/welldone-software/why-did-you-render/issues/5
